proto/api_v3/query_service.proto

```diff
@@ -40,47 +40,64 @@ message GetTraceRequest {
   bool raw_traces = 4;
 }
 
-// Query parameters to find traces. Except for num_traces, all fields should be treated
-// as forming a conjunction, e.g., "service_name='X' AND operation_name='Y' AND ...".
+// Query parameters to find traces. All fields should be treated
+// as forming a conjunction, e.g., "service_name='X' AND operation_name='Y' AND ..."
+// with the exception of `search_depth` and `raw_traces`.
+//
 // All fields are matched against individual spans, not at the trace level.
 // The returned results contain traces where at least one span matches the conditions.
-// When num_traces results in fewer traces returned, there is no required ordering.
-//
-// Note: num_traces should restrict the number of traces returned, but not all backends
-// interpret it this way. For instance, in Cassandra this limits the number of _spans_
-// that match the conditions, and the resulting number of traces can be less.
-//
-// Note: some storage implementations do not guarantee the correct implementation of all parameters.
+// There is no guaranteed ordering of the returned results.
 //
+// Note: Some storage implementations do not guarantee the correct implementation of all parameters.
 message TraceQueryParameters {
+  // `service_name` filters traces generated by a specific service.
   string service_name = 1;
+
+  // `operation_name` filters traces generated by a specific operation.
   string operation_name = 2;
 
-  // Attributes are matched against Span and Resource attributes.
-  // At least one span in a trace must match all specified attributes.
+  // `attributes` contains key-value pairs where the key is the attribute name
+  // and the value is its string representation. Attributes are matched against
+  // span and resource attributes. At least one span must match all specified attributes.
   map<string, string> attributes = 3;
 
-  // Span min start time in. REST API uses RFC-3339ns format. Required.
+  // `start_time_min` is the start of the time interval for the query. Only traces with spans
+  // that started after this time will be returned.
+  //
+  // The HTTP API uses RFC-3339ns format.
+  //
+  // This field is required.
   google.protobuf.Timestamp start_time_min = 4;
 
-  // Span max start time. REST API uses RFC-3339ns format. Required.
+  // `start_time_max` is the end of the time interval for the query. Only traces with spans
+  // that started before this time will be returned.
+  //
+  // The HTTP API uses RFC-3339ns format.
+  //
+  // This field is required.
   google.protobuf.Timestamp start_time_max = 5;
 
-  // Span min duration. REST API uses Golang's time format e.g. 10s.
+  // `duration_min` is the minimum duration of the trace. Only traces with spans that lasted
```

Comment:
> minimum duration of the trace

I don't believe this is looking at trace duration (which is ill-defined), only at individual span duration.